src/python/build_management/build_manager.py

```diff
@@ -692,7 +715,11 @@ def build_dir(self):
   def setup(self):
     """Sets up build with a particular revision."""
     self._pre_setup()
-    environment.set_value('BUILD_URL', self.build_url)
+    if self.is_auxiliary_build:
```

Comment:
why not key this off self.build_prefix? then we can avoid is_auxiliary_build. 

Also this can be made simpler if we make self.build_prefix = '' by default, and keep the 'APP_' parts while prepending.